docs/src/pages/system/the-sx-prop/the-sx-prop.md

```diff
@@ -188,6 +188,45 @@ Each property in the `sx` prop can receive a function callback as a value. This
 <Box sx={{ height: (theme) => theme.spacing(10) }} />
 ```
 
+## TypeScript usage
+
+A frequent source of confusion with the `sx` prop is TypeScript's [type widening](https://mariusschulz.com/blog/typescript-2-1-literal-type-widening), which causes this example not to work as expected:
+
+```ts
+const styles = {
+  flexDirection: 'column',
+};
+
+export default function App() {
+  return <Button sx={sx}>Example</Button>;
+}
+//    Type '{ flexDirection: string; }' is not assignable to type 'SxProps<Theme> | undefined'.
+//    Type '{ flexDirection: string; }' is not assignable to type 'CSSSelectorObject<Theme>'.
+//      Property 'flexDirection' is incompatible with index signature.
+//        Type 'string' is not assignable to type 'SystemStyleObject<Theme>'.
+```
+
+The problem is that the type of the `flexDirection` prop is inferred as `string`, which is too wide.
+To fix this, you can pass the styles object directly to the `sx` prop:
+
+```ts
+export default function App() {
+  return <Button sx={{ flexDirection: 'column' }}>Example</Button>;
+}
+```
+
+Alternatively, you can cast the object/function passed to the `sx` prop to const:
```

Comment:
Since the original code also hoisted the style, we should start with it. Let's not mix types and runtime immediately and instead view them as separate problems with regard to TypeScript quirks. In other words, the default action should be to fix the types not the runtime since that changes semantics. Especially because hoisting the style to the module scope automatically memoizes its value.

Reply:
Makes sense. Let me move this section first.